Student-Unity/src/main/java/csed/swe/studentunity/Logic/RegistrationService.java

```diff
@@ -1,37 +1,61 @@
 package csed.swe.studentunity.Logic;
 
 import csed.swe.studentunity.model.RegistrationRequest;
+import csed.swe.studentunity.model.UnverifiedUser;
 import csed.swe.studentunity.model.User;
+import csed.swe.studentunity.model.VerificationRequest;
 import lombok.AllArgsConstructor;
 import org.springframework.stereotype.Service;
 
+import java.util.Optional;
+
 @AllArgsConstructor
 @Service
 //@Transactional
 public class RegistrationService {
 
     private final UserService userService;
+    private final UnverifiedUserService unverifiedUserService;
+    private final EmailService emailService;
 
     public String addUser(RegistrationRequest request) {
         // check if email is already in use, if not: add user to database and send verification email
-        if(userService.getUser(request.getEmail()).isEmpty()) {
-            User newUser = new User(request.getEmail(), request.getStudentId(), request.getFirstName(),
-                    request.getLastName(), request.getPassword(), "student", "");
-            userService.addUser(newUser);
+        if(unverifiedUserService.getUnverifiedUser(request.getEmail()).isEmpty()
+        && userService.getUser(request.getEmail()).isEmpty()) {
+            UnverifiedUser newUser = new UnverifiedUser(request.getEmail(), request.getStudentId(), request.getFirstName(),
+                    request.getLastName(), request.getPassword(), "student");
+            String code = unverifiedUserService.addUnverifiedUser(newUser);
+            emailService.send("You have registered in Student Unity \n" +
+                    "verification code is: " + code, "verification code", request.getEmail());
             return "User registered successfully";
```

Comment:
I think that it is better to return enums instead of strings as checking for string in the API layer will be not good But it is fine we can do it later

Reply:
You are right. I intend to do that.
But I'm waiting for our next scrum to specify what should be returned.